proposals/0384-alpenglow-migration.md

```diff
@@ -0,0 +1,261 @@
+---
+simd: '0384'
+title: Alpenglow migration
+authors:
+  - Kobi Sliwinski (Anza)
+  - Ashwin Sekar (Anza)
+  - Carl Lin (Anza)
+category: Standard
+type: Core
+status: Review
+created: 2025-10-21
+feature: a1penGLz8Vm2QHYB3JPefBiU4BY3Z6JkW2k3Scw5GWP
+---
+
+## Summary
+
+Migrate to Alpenglow from TowerBFT
+
+## Motivation
+
+Migrating from TowerBFT to Alpenglow consensus requires a safe handoff
+mechanism that doesn't rollback TowerBFT confirmed user transactions.
+
+## Dependencies
+
+This proposal depends on the following accepted proposal:
+
+- **[SIMD-0326]: Alpenglow**
+
+    Requires Alpenglow to be implemented in order to migrate
+
+- **[SIMD-0307]: Add Block Footer**
+
+    Specifies `BlockMarker`, an means of disseminating metadata in a block
+
+
+## New Terminology
+
+Strong optimistic confirmation:
+
+- A block `B` is strong OC if there exists a block `T` such that `B` is the parent
+  of `T` and `slot(B) + 1 = slot(T)` and `T` contains vote transactions for `B` from
+  at least `82%` of stake.
+
+## Detailed Design
+
+### Migration Handoff
+
+1. Pick a migration boundary slot `S` as follows. Let `X` be the rooted slot in
+   which the feature flag is activated. Let the migration slot S be `X + 5000`,
+   as to avoid the beginning of an epoch. After slot `S`:
+   - Turn off packing user transactions in blocks. Any transactions not
+     belonging to the vote program will cause a block to be marked dead during
+     replay by all correct validators.
+   - Turn off optimistic confirmation/commitment reporting over RPC.
+   - Turn off rooting blocks in TowerBFT to prevent losing optimistically
+     confirmed blocks that could qualify as the Alpenglow genesis blocks.
```

Comment:
nit: could we tell users what "Turn off" means here? What would they observe differently. Would that mean `TowerSync` containing more than 32 votes?

Reply:
turn off means no roots will be set in the core code. TowerSync can never contain more than 32 votes because it's implemented as a fixed length queue, but the root distance will grow indefinitely until migration succeeds